test/AtomicFAddExt.ll

```diff
@@ -0,0 +1,118 @@
+; RUN: llvm-as < %s > %t.bc
+; RUN: llvm-spirv %t.bc --spirv-ext=+SPV_INTEL_fpga_loop_controls -o %t.spv
```

Comment:
```suggestion
; RUN: llvm-spirv %t.bc --spirv-ext=+SPV_EXT_shader_atomic_float_add -o %t.spv
```

Reply:
And I see a problem here... I assume, even with the incorrect line, the test is passing on your side, right? That is because there is no code, that checks if the extension is supported. There are two ways, how to do this:
1. Ask a SPIR-V module if it supports the extension. But I'm unsure, how it can work in your case...
2. Add `getRequiredExtension` method for an instruction class, like this:
```
llvm::Optional<ExtensionID> getRequiredExtension() const override {
  return ExtensionID::SPV_EXT_shader_atomic_float_add;
}
```

Problem here is that you are modifying already existing class (that was added for translation of core atomic instructions). So going this way requires to add another class for this particular extension.

Reply:
Whoops, thanks for noticing. It was a lazy `Ctrl + V` from another test, hence the misspell.
Good point with the extension check as well! I could probably add an `if (getOpCode() == OpAtomicFAddEXT)` check to the override you've suggested.

Reply:
Done; I've also checked that leaving the extension unspecified in the CL results in a correponding LIT failure.